ReactAndroid/src/main/java/com/facebook/react/modules/share/ShareModule.java

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright (c) 2016-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.modules.share;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.net.Uri;
+
+import com.facebook.react.bridge.JSApplicationIllegalArgumentException;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+
+/**
+ * Intent module. Launch other activities or open URLs.
+ */
+public class ShareModule extends ReactContextBaseJavaModule {
+
+  public ShareModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+  }
+
+  @Override
+  public String getName() {
+    return "ShareModule";
+  }
+
+  /**
+   * Open a chooser dialog to send text content to other apps.
+   *
+   * Refer http://developer.android.com/intl/ko/training/sharing/send.html
+   * 
+   * @param content the data to send
+   * @param title the title of the chooser dialog
+   */
+  @ReactMethod
+  public void shareText(ReadableMap content, String title, Promise promise) {
```

Comment:
Tests make it a lot easier to find breakages in code, please try to add some. It can be just a simple test that checks that we're creating the intent correctly.